### PR TITLE
[AMD-AIE] Lower zero filling linalg.fill to ukernel

### PR DESCRIPTION
**Pull request by Abhishek-Varma**
-- This commit updates `--iree-amdaie-lower-to-ukernel` pass by
   adding zero filling ukernel matcher for a linalg.fill op.

Signed-off-by: Abhishek Varma <abhvarma@amd.com>

### CODE REVIEW

**Comment by Abhishek-Varma** [comment]
For testing an e2e Matmul bf16/bf16 (input/output) for Pack-Peel, you'd need [PR 1479](https://github.com/Xilinx/mlir-aie/pull/1479).

CC: @erwei-xilinx

File: compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/Passes.cpp [hunk 1]
```diff
@@ -281,6 +281,14 @@ void addPackPeelBasedPassPipeline(OpPassManager &funcPassManager,
   funcPassManager.addPass(createHoistStaticallyBoundAllocationsPass());
   funcPassManager.addPass(createCanonicalizerPass());
 
+  // Lower to UKernels.
+  {
+    AMDAIELowerToUKernelsOptions options;
+    options.passPipeline = AIEPassPipeline::PackPeelPipeline;
```
Comment (newling) [inline compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/Passes.cpp:287]: How is the pass pipeline used in the ukernel lowering? I've tried partially removing it in https://github.com/nod-ai/iree-amd-aie/pull/345 and it works, although I haven't tried removing it from AMDAIELowerToUKernelsOptions

**Review by newling — APPROVED** [review]
LGTM, nice tests. I just had the one question about the use of the pipeline in the ukernel (Does the ukernel need to know if it's pad-pack vs pack-peel?)

File: compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/Passes.cpp (hunk 1 above)
Comment (Abhishek-Varma) [reply compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/Passes.cpp:287]: Yes, you're right - it's not needed. I forgot to remove it - thanks for pointing this out. I will do that in this PR.

Comment (Abhishek-Varma) [reply compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/Passes.cpp]: Oh. I didn't see this and have thus removed it in this PR.

Comment (Abhishek-Varma) [reply compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/Passes.cpp]: Let me know what would you prefer - i) merging just this PR or ii) the other PR followed by this.

Comment (newling) [reply compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/Passes.cpp]: (i) makes sense, I've closed the other PR now

**Pull request merged**